src/NHibernate/Dialect/SybaseSQLAnywhere10Dialect.cs

```diff
@@ -63,17 +63,17 @@ public SybaseSQLAnywhere10Dialect()
 
 		protected virtual void RegisterCharacterTypeMappings()
 		{
-			RegisterColumnType(DbType.AnsiStringFixedLength, "CHAR(1)");
+			RegisterColumnType(DbType.AnsiStringFixedLength, "CHAR(255)");
 			RegisterColumnType(DbType.AnsiStringFixedLength, 32767, "CHAR($l)");
-			RegisterColumnType(DbType.AnsiString, "VARCHAR(1)");
+			RegisterColumnType(DbType.AnsiString, "VARCHAR(255)");
 			RegisterColumnType(DbType.AnsiString, 32767, "VARCHAR($l)");
 			RegisterColumnType(DbType.AnsiString, 2147483647, "LONG VARCHAR");
-			RegisterColumnType(DbType.StringFixedLength, "NCHAR(1)");
+			RegisterColumnType(DbType.StringFixedLength, "NCHAR(255)");
 			RegisterColumnType(DbType.StringFixedLength, 32767, "NCHAR($l)");
-			RegisterColumnType(DbType.String, "NVARCHAR(1)");
+			RegisterColumnType(DbType.String, "NVARCHAR(255)");
 			RegisterColumnType(DbType.String, 32767, "NVARCHAR($l)");
 			RegisterColumnType(DbType.String, 2147483647, "LONG NVARCHAR");
-			RegisterColumnType(DbType.Binary, "BINARY(1)");
+			RegisterColumnType(DbType.Binary, "BINARY(8000)");
```

Comment:
Huh, I understand now why basic tests were failing with string truncation now... But this fix deserves a Jira on its own.

I have created [NH-4046](https://nhibernate.jira.com/browse/NH-4046). I think it should be fixed at least as a separated commit, maybe even a separated PR. `SybaseASE15Dialect` has the issue too.

We should be able to use `SybaseASA9Dialect` default lengths safely, since this is an old name for Anywhere. In this regard, `Binary(8000)` is maybe too big. `SybaseASA9Dialect` declares `Binary(255)` instead.